tool_path_planner/src/plane_slicer_raster_generator.cpp

```diff
@@ -453,10 +452,43 @@ boost::optional<ToolPaths> PlaneSlicerRasterGenerator::generate()
   {
     CONSOLE_BRIDGE_logDebug("%s No mesh data has been provided", getName().c_str());
   }
-  // computing mayor axis using oob
-  vtkSmartPointer<vtkOBBTree> oob = vtkSmartPointer<vtkOBBTree>::New();
-  Vector3d corner, x_dir, y_dir, z_dir, sizes;  // x is longest, y is mid and z is smallest
-  oob->ComputeOBB(mesh_data_, corner.data(), x_dir.data(), y_dir.data(), z_dir.data(), sizes.data());
+  // Assign the longest axis of the bounding box to x, middle to y, and shortest to z.
+  Vector3d corner, x_dir, y_dir, z_dir, sizes;
+  if (config_.raster_wrt_global_axes)
+  {
+    // Determine extent of mesh along axes of current coordinate frame
+    VectorXd bounds(6);
+    std::vector<Vector3d> extents;
+    mesh_data_->GetBounds(bounds.data());
+    extents.push_back(Vector3d::UnitX() * (bounds[1] - bounds[0]));  // Extent in x-direction of supplied mesh
+                                                                     // coordinate frame
+    extents.push_back(Vector3d::UnitY() * (bounds[3] - bounds[2]));  // Extent in y-direction of supplied mesh
+                                                                     // coordinate frame
+    extents.push_back(Vector3d::UnitZ() * (bounds[5] - bounds[4]));  // Extent in z-direction of supplied mesh
```

Comment:
It might be good to call cwise.abs() to ensure that the values of extends only have positive values even though theoretically it should always be positive